src/components/SQForm/SQFormResetButtonWithConfirmation.tsx

```diff
@@ -1,8 +1,24 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {useDialog} from '@selectquotelabs/sqhooks';
 import {DialogAlert, RoundedButton} from 'scplus-shared-components';
 import {useFormButton} from './useFormButton';
+import {useDialog} from '@selectquotelabs/sqhooks';
+
+type SQFormResetButtonWithConfirmationTypes = {
+  /** The contents of the form button; usually text */
+  children: React.ReactNode;
+  /** Title of the confirmation dialog */
+  confirmationTitle?: string;
+  /** Content of the confirmation dialog */
+  confirmationContent: React.ReactNode;
+  /** Whether the button is disabled */
+  isDisabled?: boolean;
+  /** Title of the button */
+  buttonTitle?: string;
+  /** Callback for reset*/
+  onReset?: () => void;
+  /** style variant, "contained" or "outlined"*/
+  variant?: string;
```

Comment:
`variant` should be limited to "contained" or "outlined" if those are the only options allowed
```suggestion
  variant?: 'contained' | 'outlined';
```

Reply:
ahh, good catch! a union type would work perfectly!